### PR TITLE
Add PhpdocVarAnnotationCorrectOrderFixer

### PR DESCRIPTION
**Pull request by kubawerlos**
Closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2159

### CODE REVIEW

File: tests/Fixer/Phpdoc/PhpdocVarAnnotationCorrectOrderFixerTest.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        ];
+
+        yield [
+            '<?php
+/** @var Foo|Bar|mixed|int $someWeirdLongNAME__123 */
+',
+            '<?php
+/** @var $someWeirdLongNAME__123 Foo|Bar|mixed|int */
+',
+        ];
```
Comment (ntzm) [inline tests/Fixer/Phpdoc/PhpdocVarAnnotationCorrectOrderFixerTest.php]: Please add cases for changes that are not expected:

```
/** @var */
/** @var $foo */
/** @var Bar */
```

And multi-line descriptions:

```
/**
 * @var $bar Foo long description
 *               goes here
 */
```

**Comment by dmvdbrugge** [comment]
What is the reasoning behind not fixing `@param`?

**Comment by kubawerlos** [comment]
@dmvdbrugge the alternative for 2 is not a acceptable as we cannot assume some will use `PhpdocNoAliasTagFixer`.

`@param` is not fixed as the idea is to fix `@var` cases - I've never seen `@param` with variable name before type and `@param` more often has description than `@var`.

I'm not sure now, maybe we can aim to have `PhpdocCorrectOrderOfTypeAndNameFixer` to handle all Phpdoc types that have type and name the same, yet I prefer to add [`mixed`](https://github.com/kubawerlos/php-cs-fixer-custom-fixers/blob/v1.5.1/tests/Fixer/PhpdocParamTypeFixerTest.php#L108) when type is missing instead of getting first word of string behind the name as it can do more harm like:

```diff
 /**
- * @param $foo This is foo
+ * @param This $foo us foo
  */

**Comment by dmvdbrugge** [comment]
> the alternative for 2 is not a acceptable as we cannot assume some will use `PhpdocNoAliasTagFixer`.

Was thinking that as well, removed the suggestion.

As for the `@param`, makes sense. So only those 2 points for me.

**Comment by kubawerlos** [comment]
@dmvdbrugge `@type` added, yet I cannot find valid case for priority integration test with `PhpdocVarWithoutNameFixer` - it handles well name with both orders:
```diff
 <?php
 /**
- * @var FooInterface $foo
+ * @var FooInterface
  */
 $foo;
 
 /**
- * @var $bar BarInterface
+ * @var BarInterface
  */
 $bar;
```

File: tests/Fixer/Alias/NoAliasFunctionsFixerTest.php [hunk 2]
```diff
@@ -40,7 +40,7 @@ public function provideFixCases()
         $cases = [];
 
         foreach (['internalSet', 'imapSet'] as $setStaticAttributeName) {
-            /** @var $aliases string[] */
+            /** @var string[] $aliases */
```
Comment (dmvdbrugge) [inline tests/Fixer/Alias/NoAliasFunctionsFixerTest.php]: So nice that it immediately adds value to the project itself 🎉

**Comment by keradus** [comment]
Thank you @kubawerlos.

**Comment by phansys** [comment]
Thank you @kubawerlos! :tada:

**Pull request merged**